### DISCUSSION TITLE
Optimistic nature of the code

### DISCUSSION THREAD

**Discussion by kant2002** [dotnet/winforms#7132] [Q&A]
I again looking at the code around places where I touch things
https://github.com/dotnet/winforms/blob/58159f3e27c336f468302c67da87b3b3f55c90c9/src/System.Windows.Forms/src/System/Windows/Forms/AccessibleObject.EnumVariantObject.cs#L46-L61
and 
https://github.com/dotnet/winforms/blob/58159f3e27c336f468302c67da87b3b3f55c90c9/src/System.Windows.Forms/src/System/Windows/Forms/AccessibleObject.EnumVariantObject.cs#L120

but HRESULT is ignored
https://github.com/dotnet/winforms/blob/58159f3e27c336f468302c67da87b3b3f55c90c9/src/System.Windows.Forms.Primitives/src/Interop/OleAut32/Interop.IEnumVariant.cs#L17-L27

Isn't this too brave?

**Comment by RussKie** [comment]
It's likely an oversight. There are a few gaps in our interops, e.g., https://github.com/dotnet/winforms/issues/3151.
